apps/insights/src/services/clickhouse.ts

```diff
@@ -354,20 +386,23 @@ export const getHistoricalPrices = async ({
     ),
     {
       query: `
-          SELECT toUnixTimestamp(time) AS timestamp, avg(price) AS price, avg(confidence) AS confidence
+          SELECT toUnixTimestamp(toStartOfInterval(publishTime, INTERVAL ${resolution})) AS timestamp, avg(price) AS price, avg(confidence) AS confidence
           FROM prices
-          WHERE cluster = 'pythnet'
-          AND symbol = {symbol: String}
-          AND version = 2
-          AND time > fromUnixTimestamp(toInt64({until: String})) - INTERVAL 5 MINUTE
-          AND time < fromUnixTimestamp(toInt64({until: String}))
-          AND publisher = ''
-          GROUP BY time
-          ORDER BY time ASC
+          PREWHERE
```

Comment:
Hmm I'm not sure I understand the reason for splitting the where clause and using prewhere here, does postgres not optimize this query well?

Reply:
to really get to the bottom here we should probably measure a couple different queries and see which one is faster. 

that said, the `(cluster, publisher, symbol)` combo follows the specified physical sort order of the table, and so forcing the engine to filter by that first _should_ be some kind of guarantee that we're always doing lookups as fast as possible.

i could be wrong but in classic postgres i don't think there's any way of telling the engine exactly how to use indices even if we _know_ which way is faster.